src/plugins/mpi/pmix/pmixp_agent.c

```diff
@@ -291,6 +342,59 @@ static void *_pmix_timer_thread(void *unused)
 	return NULL;
 }
 
+/*
+ * thread for codes from abort
+ */
+static void *_pmix_abort_thread(void *args)
+{
+	PMIXP_DEBUG("Start abort thread");
+	eio_handle_mainloop(_abort_handle);
+	PMIXP_DEBUG("Abort thread exit");
+}
+
+int pmixp_abort_agent_start(char ***env)
+{
+	int abort_server_socket = -1;
+	if ((abort_server_socket = slurm_init_msg_engine_port(0)) < 0) {
+		PMIXP_ERROR("Error slurm_open_stream %s", strerror(errno));
+		return SLURM_COMMUNICATIONS_CONNECTION_ERROR;
+	}
+
+	slurm_addr_t abort_server;
+	memset(&abort_server, 0, sizeof(slurm_addr_t));
+
+	slurm_get_stream_addr(abort_server_socket, &abort_server);
+	PMIXP_DEBUG("Abort server ip:port: %s:%d", inet_ntoa(abort_server.sin_addr), abort_server.sin_port);
+
+	char ip_buffer[INET_ADDRSTRLEN];
+	inet_ntop(AF_INET, &abort_server.sin_addr, ip_buffer, sizeof(ip_buffer)); // TODO: check error return
+
+	setenvf(env, PMIXP_SLURM_ABORT_AGENT_IP, "%s", ip_buffer);
+	setenvf(env, PMIXP_SLURM_ABORT_AGENT_PORT, "%d", abort_server.sin_port);
+
+	eio_obj_t *obj;
+	_abort_handle = eio_handle_create(0);
+	obj = eio_obj_create(abort_server_socket, &abort_ops, (void *)(-1));
+	eio_new_initial_obj(_abort_handle, obj);
+
+	slurm_thread_create(&_abort_tid, _pmix_abort_thread, (void*)abort_server_socket);
```

Comment:
`abort_server_socket` passed as an argument to `_pmix_abort_thread`:
https://github.com/BorisBochkaryov/slurm/blob/bf07f61707d99fc43757f51715603c9f020a4996/src/plugins/mpi/pmix/pmixp_agent.c#L380
but not used:
https://github.com/BorisBochkaryov/slurm/blob/bf07f61707d99fc43757f51715603c9f020a4996/src/plugins/mpi/pmix/pmixp_agent.c#L348-L353